src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -545,15 +545,17 @@ protected async Task<HostingEnvironment> GetHostingEnvironment(string subscripti
                 case DiagnosticStampType.ASEV2:
                     hostingEnv.HostingEnvironmentType = HostingEnvironmentType.V2;
                     break;
-
+                case DiagnosticStampType.ASEV3:
+                    hostingEnv.HostingEnvironmentType = HostingEnvironmentType.V3;
+                    break;
                 default:
                     hostingEnv.HostingEnvironmentType = HostingEnvironmentType.None;
                     break;
             }
 
             string stampName = !string.IsNullOrWhiteSpace(hostingEnv.InternalName) ? hostingEnv.InternalName : hostingEnv.Name;
 
-            if (platformType==null && (stampPostBody.Kind == DiagnosticStampType.ASEV1 || stampPostBody.Kind == DiagnosticStampType.ASEV2))
+            if (platformType==null && (stampPostBody.Kind == DiagnosticStampType.ASEV1 || stampPostBody.Kind == DiagnosticStampType.ASEV2 || stampPostBody.Kind == DiagnosticStampType.ASEV3))
```

Comment:
nitpick : how about updating this if clause to check for (platformType == null && hostingEnv.HostingEnvironmentType != HostingEnvironmentType.None)
(refer to line 552 above)